### DISCUSSION TITLE
"Channel Discovery Assistant" built into Search & Channel Switcher

### DISCUSSION THREAD

**Discussion by sadohert** [mattermost/community#38] [Practical Ideas]
Channel discovery can often create "moments of pause" for Mattermost end-users:

"Where should I ask my question?"

"How do I find the people who know about Docker?"

"Does anyone discuss their favourite movies and TV shows on here?"

The moments impact the users "flow" in the app.  Overtime this problem will work itself out as the user becomes familiar with the channels available to them, but the brand new users won't have a full grasp on the channels they care about until their colleagues naturally bring them into the right ones.

Search is great for targeting specific discussions that relate to the search terms, but won't necessarily help the user discover other beneficial sources of information/discussion.

For example, you could search for "Docker" and find a tonne of recent results embedded in a specific teams channel discussion, but this may not bring up the company-wide "Docker" channel where all the subject-matter experts have related discussion.

Ideally channel names (or naming conventions) would make it easier for a user to discovery relevant channels, but  could we do better?  Could the "Search Results" panel also have a "Top Channels" summary tab that sorts the top 5 channels with relevant results?

**Comment by esethna** [comment]
I think a Mattermost messaging home screen is something we've talked about along these lines. For example,

- ongoing meeting in channel x, join it?
- There's a lot of conversation happening in channel y, check it out
- your collegues are in channel z, join it? 
- digest of unreads that you missed since last activity
etc.

**Reply by sadohert** [reply]
Nice.  Some intelligence on those things would be awesome!

**Comment by Willyfrog** [comment]
I'm in a discord server where someone wrote a bit that ask you questions. You reply to those questions by reacting with an emoji, and based on your responses it gets you into different channels that you showed interest. Given this discord channel has about 50 channels or so of different themes, it is an interesting way to start with a nice set of channels that interests you.
